filebeat/module/traefik/access/ingest/pipeline.json

```diff
@@ -5,19 +5,21 @@
             "grok": {
                 "field": "traefik.access.message",
                 "patterns": [
-                    "(?:%{NUMBER:traefik.access.body_sent.bytes}|-)( \"%{DATA:traefik.access.referrer}\")?( \"%{DATA:traefik.access.agent}\")?(?:%{NUMBER:traefik.access.request_count}|-)?( \"%{DATA:traefik.access.frontend_name}\")?( \"%{DATA:traefik.access.backend_url}\")?"
+                    "%{NUMBER:traefik.access.body_sent.bytes:int}( \"%{GREEDYDATA:traefik.access.referrer}\" \"%{DATA:traefik.access.agent}\" %{NUMBER:traefik.access.request_count:int} \"%{DATA:traefik.access.frontend_name}\" \"%{DATA:traefik.access.backend_url}\" %{NUMBER:traefik.access.duration:int}ms)?"
```

Comment:
I see most `}|-)?` parts are removed. Are these not needed anymore?

Reply:
Good question. At the beginning I thought that Traefik handles missing fields by replacing them with a `-`, it seems to be like this for some fields. Then I investigate more and I couldn't find any place where they were actually explaning that they were doing this for every field so I left only the ones that I was sure that they were handle this way.

Reply:
The part I worry here is, if they use `-` for the fields where they were removed, a user upgrading to the new version it will break. I'm ok removing it if we are 100% confident they are not used, all the ones we are not 100% sure should stay in.